runtimeconfig/manager.go

```diff
@@ -147,26 +150,51 @@ func (om *Manager) loop(ctx context.Context) error {
 // loadConfig loads all configuration files using the loader function then merges the yaml configuration files into one yaml document.
 // and notifies listeners if successful.
 func (om *Manager) loadConfig() error {
-	mergedConfig := map[string]interface{}{}
+	rawData := map[string][]byte{}
+	hashes := map[string]string{}
+
 	for _, f := range om.cfg.LoadPath {
-		yamlFile := map[string]interface{}{}
 		buf, err := os.ReadFile(f)
 		if err != nil {
 			om.configLoadSuccess.Set(0)
 			return errors.Wrapf(err, "read file %q", f)
 		}
-		err = yaml.Unmarshal(buf, &yamlFile)
+
+		rawData[f] = buf
+		hashes[f] = fmt.Sprintf("%x", sha256.Sum256(buf))
+	}
+
+	// check if new hashes are the same as before
+	sameHashes := true
+	for f, h := range hashes {
+		if om.fileHashes[f] != h {
```

Comment:
Isn't `om.fileHashes` initially `nil`?

Reply:
Yes. Reading from nil map is fine.

Reply:
@pstibrany you're right, I tested it in the meantime. I must be remembering it wrong, I thought reading from nil maps would crash.